src/api/ApiResponder.php

```diff
@@ -144,7 +109,7 @@ protected function get_interface_response_createGame($interface, $args) {
         if (isset($retval)) {
             foreach ($playerIdArray as $playerId) {
                 if (isset($playerId)) {
-                    $interface->update_last_action_time($playerId, $retval['gameId']);
+                    $interface->player()->update_last_action_time($playerId, $retval['gameId']);
```

Comment:
Oh, i see.  Yeah, okay.

So the alternative i was thinking of is not to have a `select_interface_type()` function, and instead generally return a BMInterface, and on line 100 above use `$interface->player()->create_game()`.

I don't think i feel strongly --- it seems a little inconsistent that on line 100 you're saying "well, we know this is a BMInterfacePlayer object" and on line 112 in the same function you're saying "we can't make assumptions about what kind of BMInterface object this is".  But i don't think it's a big deal.